Frameworks/CoreText/CTTypesetter.mm

```diff
@@ -92,22 +90,23 @@ CTLineRef CTTypesetterCreateLineWithOffset(CTTypesetterRef ts, CFRange range, do
 
 /**
  @Status Interoperable
- @Notes
 */
 CFIndex CTTypesetterSuggestLineBreak(CTTypesetterRef typesetter, CFIndex startIndex, double width) {
     return CTTypesetterSuggestLineBreakWithOffset(typesetter, startIndex, width, 0.0f);
 }
 
 /**
  @Status Interoperable
- @Notes
 */
-CFIndex CTTypesetterSuggestLineBreakWithOffset(CTTypesetterRef ts, CFIndex index, double width, double offset) {
-    _CTTypesetter* typesetter = static_cast<_CTTypesetter*>(ts);
-    _CTFrame* frame = _DWriteGetFrame(static_cast<CFAttributedStringRef>(typesetter->_attributedString.get()),
-                                      CFRangeMake(index, [typesetter->_string length] - index),
+CFIndex CTTypesetterSuggestLineBreakWithOffset(CTTypesetterRef typesetter, CFIndex index, double width, double offset) {
+    _CTFrame* frame = _DWriteGetFrame(typesetter->AttributedString(),
+                                      CFRangeMake(index, CFAttributedStringGetLength(typesetter->AttributedString()) - index),
                                       CGRectMake(offset, 0, width, FLT_MAX));
-    return ([frame->_lines count] > 0) ? static_cast<_CTLine*>([frame->_lines firstObject])->_strRange.length : 0;
+    if ([frame->_lines count] > 0) {
```

Comment:
>[frame->_lines count] [](start = 8, length = 21)

These objc calls can already be changed to CFCalls due to bridging so you dont have to touch this again when you fix CTLines